### PR TITLE
CodeGen: Port ExpandLargeDivRem to new pass manager

### PR DESCRIPTION
**Pull request by arsenm**

### CODE REVIEW

File: llvm/lib/Passes/PassRegistry.def [hunk 1]
```diff
@@ -445,6 +445,7 @@ FUNCTION_PASS("transform-warning", WarnMissedTransformationsPass())
 FUNCTION_PASS("tsan", ThreadSanitizerPass())
 FUNCTION_PASS("memprof", MemProfilerPass())
 FUNCTION_PASS("declare-to-assign", llvm::AssignmentTrackingPass())
+FUNCTION_PASS("expand-large-div-rem", ExpandLargeDivRemPass(TM));
```
Comment (arsenm) [inline llvm/lib/Passes/PassRegistry.def:448]: This seems to already exist without the constructor argument in MachinePassRegistry.def, not sure what's going on there

**Review by mgehre-amd — APPROVED** [review]
Nice, thanks!

File: llvm/lib/CodeGen/ExpandLargeDivRem.cpp [hunk 2]
```diff
@@ -128,6 +129,13 @@ class ExpandLargeDivRemLegacyPass : public FunctionPass {
 };
 } // namespace
 
+PreservedAnalyses ExpandLargeDivRemPass::run(Function &F,
+                                             FunctionAnalysisManager &FAM) {
+  const TargetSubtargetInfo *STI = TM->getSubtargetImpl(F);
+  PreservedAnalyses PA = PreservedAnalyses::none();
```
Comment (aeubanks) [inline llvm/lib/CodeGen/ExpandLargeDivRem.cpp:135]: inline this

File: llvm/lib/Passes/PassRegistry.def (hunk 1 above)
Comment (aeubanks) [reply llvm/lib/Passes/PassRegistry.def:448]: what do you mean?

```
$ ./build/rel/bin/opt -p expand-large-div-rem -S /dev/null
./build/rel/bin/opt: unknown pass name 'expand-large-div-rem'
```

**Review by aeubanks — APPROVED** [review]

File: llvm/lib/Passes/PassRegistry.def (hunk 1 above)
Comment (arsenm) [reply llvm/lib/Passes/PassRegistry.def:448]: I mean this exists: https://github.com/llvm/llvm-project/blob/7b5505b0d597eddadd5ffd4bbf80a7e8804ec809/llvm/include/llvm/CodeGen/MachinePassRegistry.def#L46

**Pull request merged**